packages/experience/src/hooks/use-navigate-with-search-params.ts

```diff
@@ -0,0 +1,58 @@
+import { conditional } from '@silverhand/essentials';
+import { useCallback } from 'react';
+import {
+  type NavigateFunction,
+  type NavigateOptions,
+  type To,
+  useNavigate,
+} from 'react-router-dom';
+
+import { searchKeys } from '@/utils/search-parameters';
+
+const useNavigateWithSearchParams = () => {
```

Comment:
Need to add some IntelliSense comments to explain why we need this, and what the outcome would be.